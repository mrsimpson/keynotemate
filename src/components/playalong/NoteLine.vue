```vue
<template>
  <div class="note-line-container">
    <div class="note-line">
      <slot name="notes"></slot>
    </div>
    <div class="lyrics-container">
      <slot name="lyrics"></slot>
    </div>
  </div>
</template>

<style scoped>
.note-line-container {
  display: flex;
  flex-direction: column;
  gap: var(--spacing-md);
}

.note-line {
  display: flex;
  gap: var(--spacing-lg);
  padding: var(--spacing-lg) 0;
  min-height: 60px;
}

.lyrics-container {
  display: flex;
  gap: var(--spacing-lg);
  min-height: 24px;
  padding: 0 0 var(--spacing-md);
}
</style>
```