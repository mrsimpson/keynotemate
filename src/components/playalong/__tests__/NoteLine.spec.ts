import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import NoteLine from '@/components/playalong/NoteLine.vue'

describe('NoteLine Component', () => {
  it('renders note line snapshot', () => {
    const wrapper = mount(NoteLine, {
      slots: {
        notes: '<div>Note 1</div><div>Note 2</div>',
        lyrics: '<div>Lyric 1</div><div>Lyric 2</div>',
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders empty note line snapshot', () => {
    const wrapper = mount(NoteLine)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
