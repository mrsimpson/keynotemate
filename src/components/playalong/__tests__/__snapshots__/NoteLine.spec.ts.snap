// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NoteLine Component > renders empty note line snapshot 1`] = `
"<div data-v-1eaf1b14="" class="note-line-container">
  <div data-v-1eaf1b14="" class="note-line"></div>
  <div data-v-1eaf1b14="" class="lyrics-container"></div>
</div>"
`;

exports[`NoteLine Component > renders note line snapshot 1`] = `
"<div data-v-1eaf1b14="" class="note-line-container">
  <div data-v-1eaf1b14="" class="note-line">
    <div>Note 1</div>
    <div>Note 2</div>
  </div>
  <div data-v-1eaf1b14="" class="lyrics-container">
    <div>Lyric 1</div>
    <div>Lyric 2</div>
  </div>
</div>"
`;
